armi/physics/neutronics/globalFlux/tests/test_globalFluxInterface.py

```diff
@@ -0,0 +1,150 @@
+        self.r.core.p.keff = 1.00
+
+    def interactEveryNode(self, cycle, node):
+        gi.GlobalFluxInterface.interactEveryNode(self, cycle, node)
+        self.r.core.p.keff = 1.01
+
+
+class MockGlobalFluxWithExecuters(gi.GlobalFluxInterfaceUsingExecuters):
+    def getExecuterCls(self):
+        return MockGlobalFluxExecuter
+
+
+class MockGlobalFluxExecuter(gi.GlobalFluxExecuter):
+    def _readOutput(self):
+        class MockOutputReader:
+            def apply(self, r):
+                r.core.p.keff += 0.01
+
+        return MockOutputReader()
+
+
+class TestGlobalFluxOptions(unittest.TestCase):
+    def test_readFromSettings(self):
+        cs = settings.Settings()
+        opts = gi.GlobalFluxOptions("neutronics-run")
+        opts.fromUserSettings(cs)
+        self.assertFalse(opts.adjoint)
+
+    def test_readFromReactors(self):
+        reactor = MockReactor()
+        opts = gi.GlobalFluxOptions("neutronics-run")
+        opts.fromReactor(reactor)
+        self.assertEqual(opts.geomType, "spiral")
+
+
+class TestGlobalFluxInterface(unittest.TestCase):
+    def test_interaction(self):
+        """
+        Ensure the basic interaction hooks work
+
+        Check that a 1000 pcm rx swing is observed due to the mock.
+        """
+        cs = settings.Settings()
+        o, r = test_reactors.loadTestReactor()
+        gfi = MockGlobalFluxInterface(r, cs)
+        gfi.interactBOC()
+        gfi.interactEveryNode(0, 0)
+        gfi.interactEOC()
+        self.assertAlmostEqual(r.core.p.rxSwing, 1000)
+
+    def test_getIOFileNames(self):
+        cs = settings.Settings()
+        gfi = MockGlobalFluxInterface(MockReactor(), cs)
+        inf, _outf, _stdname = gfi.getIOFileNames(1, 2, 1)
+        self.assertEqual(inf, "armi001_2_001.GlobalFlux.inp")
+
+
+class TestGlobalFluxInterfaceWithExecuters(unittest.TestCase):
+    """Tests for the default global flux execution."""
+
+    def test_executerInteraction(self):
+        cs = settings.Settings()
+        _o, r = test_reactors.loadTestReactor()
+        r.core.p.keff = 1.0
+        gfi = MockGlobalFluxWithExecuters(r, cs)
+        gfi.interactBOC()
+        gfi.interactEveryNode(0, 0)
+        r.p.timeNode += 1
+        gfi.interactEveryNode(0, 1)
+        gfi.interactEOC()
+        self.assertAlmostEqual(r.core.p.rxSwing, (1.02 - 1.01) / 1.01 * 1e5)
+
+
+class TestGlobalFluxResultMapper(unittest.TestCase):
+    def test_mapper(self):
+        MCC2_SETTINGS = {"xsKernel": "MC2v2"}  # compat with ISOAA test lib
+        _o, r = test_reactors.loadTestReactor(customSettings=MCC2_SETTINGS)
+        applyDummyFlux(r)
+        r.core.lib = isotxs.readBinary(ISOAA_PATH)
+        mapper = gi.GlobalFluxResultMapper()
+        mapper.r = r
+        mapper._renormalizeNeutronFluxByBlock(100)
+        self.assertAlmostEqual(r.core.calcTotalParam("power", generationNum=2), 100)
+
+        mapper._updateDerivedParams()
+        self.assertGreater(r.core.p.maxPD, 0.0)
```

Comment:
You should be able to calculate the expected values for all this stuff given the knowledge of the number of blocks in the core (using len(core.getBlocks()) so there is no maintenance burden as the test reactor changes). Testing assertGreater doesn't really prove much, just that it is set to a float > 0. 

For example, I could change _renormalizeNeutronFluxByBlock to set everything to 1 or divide by powerRatio and all these test would likely still pass.

Reply:
This is just a test that the mapping process operates, not that it is right. These tests do not intend to test physics of any kind. Areal pd here is a summation over blocks of a param set by physics modules that do not exist in the framework. The summation of params is tested elsewhere.